book/visual-effects.md

```diff
@@ -1638,23 +1558,37 @@ the edge of the blur will not be sharp.
 
 [mdn-blur]: https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/blur()
 
+``` {.python}
+class AlphaAndBlend:
+    def __init__(self, alpha, blend_mode, children):
+        self.alpha = alpha
+        self.blend_mode = parse_blend_mode(blend_mode)
+        self.children = children
+        self.rect = skia.Rect.MakeEmpty()
+        for cmd in self.children:
+            self.rect.join(cmd.rect)
+
+    def execute(self, canvas):
+        paint = skia.Paint(Alphaf=self.alpha,
+            BlendMode=self.blend_mode)
+        should_save = self.alpha < 1 or \
+            self.blend_mode != skia.BlendMode.kSrcOver
+        if should_save:
+            canvas.saveLayer(paint=paint)
+        for cmd in self.children:
+            cmd.execute(canvas)
+        if should_save:
+            canvas.restore()
+```
+
+Then we'd use `AlphaAndBlend` instead of using `Alpha` and `Blend`
+separately:
```

Comment:
Is the result actually different than Alpha and Blend separately? If not, is it worth the effort to combine them in the chapter?

Reply:
You get one layer instead of two, so you still save by combining.

Reply:
Specifically if there is both alpha and also blend.